tools/apiview/parsers/cpp-api-parser/ApiViewProcessor/ApiViewDiagnostic.hpp

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+#pragma once
+#include <string>
+
+struct ApiViewDiagnostic
+{
+  enum class DiagnosticLevel
+  {
+    None = 0,
+    Info = 1,
+    Warning = 2,
+    Error = 3
+  };
+  std::string DiagnosticId;
+  std::string HelpLinkUri;
+  std::string TargetId;
+  DiagnosticLevel Level;
+};
+
+enum class ApiViewDiagnostics
```

Comment:
When I see diagnostics I think of a system that will give me more inforamtion on what is happening in the tool.  In this case the diagnostic is not about the tool but rather is about the code we are parsing.

We should use a name to reflect this isn't about diagnosing the tool, perhaps `ApiViewParserResult`.

NOTE: This change is probably huge and can be done in a follow up change

Reply:
An interesting insight. ApiView considers "Diagnostics" to be issues in the code being reviewed (that's literally how it's expressed in the schema). 

Right now there are no diagnostics issued (I'm literally working on coding the first one right now), so it's relatively easy to change the name.

Reply:
`*Error` doesn't sound right since the level specifies it isn't always an Error.